cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1463,4 +1430,40 @@ protected Future<Boolean> delete(Reconciliation reconciliation) {
     protected KafkaRebalanceStatus createStatus(KafkaRebalance ignored) {
         return new KafkaRebalanceStatus();
     }
+
+    /**
+     * Create a watch on {@code KafkaRebalance} in the given {@code namespace}.
+     *
+     * @param namespace The namespace to watch, or "*" to watch all namespaces.
+     * @param onClose Callback called when the watch is closed.
+     *
+     * @return A future which completes when the watcher has been created.
+     */
+    @Override
+    public Future<Watch> createWatch(String namespace, Consumer<WatcherException> onClose) {
+        watcher =  new OperatorWatcher<KafkaRebalance>(this, namespace, onClose) {
+            public void eventReceived(Action action, KafkaRebalance kafkaRebalance) {
+                Reconciliation reconciliation = new Reconciliation("kafkarebalance-watch", kafkaRebalance.getKind(),
+                        kafkaRebalance.getMetadata().getNamespace(), kafkaRebalance.getMetadata().getName());
+
+                LOGGER.debugCr(reconciliation, "EventReceived {} on {} with status [{}] and {}={}", action,
+                        kafkaRebalance.getMetadata().getName(),
+                        kafkaRebalance.getStatus() != null ? rebalanceStateConditionType(kafkaRebalance.getStatus()) : null,
+                        ANNO_STRIMZI_IO_REBALANCE, rawRebalanceAnnotation(kafkaRebalance));
+
+                withLock(reconciliation, LOCK_TIMEOUT_MS,
+                        () -> reconcileRebalance(reconciliation, action == Action.DELETED ? null : kafkaRebalance));
+            }
+        };
+        return VertxUtil.async(vertx, () -> resourceOperator.watch(namespace, selector(), watcher));
+    }
+
+    /**
+     * Get watcher.
+     *
+     * @return watcher.
+     */
+    public Watcher getWatcher() {
+        return watcher;
+    }
```

Comment:
Same as above, why do you have to override the createWatch method? That breaks the abstraction.

Reply:
The inherited `createWatch` method was not being used by the `KafkaRebalanceAssemblyOperator` class, `createRebalanceWatch` was used in its place with some extra logic. Since `createRebalanceWatch` in `KafkaRebalanceAssemblyOperator` is doing the same thing as `createWatch` method in cousin methods e.g.  `kafkaAssemblyOperator` and `kafkaMirrorMakerAssemblyOperator` it seemed like it would make more sense if we removed `createRebalanceWatch`  in favor of `createWatch` with the extra logic there. 

This allows changes like the following in the ClusterOperator class, adding kafkaRebalanceAssemblyOperator to the list of operators instead of treating it differently from the other operators:
```
      if (!config.isPodSetReconciliationOnly()) {
            List<AbstractOperator<?, ?, ?, ?>> operators = new ArrayList<>(asList(
                    kafkaAssemblyOperator, kafkaMirrorMakerAssemblyOperator,
                    kafkaConnectAssemblyOperator, kafkaBridgeAssemblyOperator, kafkaMirrorMaker2AssemblyOperator,
                    kafkaRebalanceAssemblyOperator));
            for (AbstractOperator<?, ?, ?, ?> operator : operators) {
                startFutures.add(operator.createWatch(namespace, operator.recreateWatch(namespace)).compose(w -> {
                  ...
            }
```

Reply:
But why not use the original `createWatch`? This way, with its own implementation, it is not like the other operators, bu quite hidden.

Reply:
I figured that the extra logic taken from the `createRebalanceWatch` may be important so didn't want to remove it

Reply:
What is the extra logic? Is it really needed? The point of the abstraction is to try to use the same method. Not to override it everywhere - that just makes it harder to find the differences and the places which for example need to be updated. So maybe its better to have a special method for it and have it clear that it is different.

Reply:
> What is the extra logic? 

Just the evenReceived() method changes in the diff above 

> Is it really needed?

Not sure exactly how important it is, I'll take a deeper look into why they were added

> Not to override it everywhere - that just makes it harder to find the differences and the places which for example need to be updated. So maybe its better to have a special method for it and have it clear that it is different.

I understand, I figured overriding it just here and the AbstractConnectAssemblyOperator would be better than having special methods e.g. we know that createWatch() is the method we need to search for to know how the watch is created across operators even if a couple have slightly different implementations